atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/workflow/ring/RingGraph.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.workload.workflow.ring;
+
+import com.google.common.collect.Iterators;
+import com.palantir.logsafe.Preconditions;
+import com.palantir.logsafe.SafeArg;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import one.util.streamex.EntryStream;
+import one.util.streamex.IntStreamEx;
+
+public final class RingGraph {
+    private final Map<Integer, Optional<Integer>> ring;
+
+    private RingGraph(Map<Integer, Optional<Integer>> ring) {
+        this.ring = ring;
+    }
+
+    public static RingGraph init(int size) {
```

Comment:
`init` seems to be used only in tests. Would a test util method using the constructor above be better?

Reply:
Happy with either-or, but purposefully left it here as it's a valid construct of the interface (i.e. give me an _empty_ ring for these nodes)